### ISSUE TITLE
Introduce an Upgrade BDD feature

### ISSUE THREAD

**Issue by jbertran** [scality/metalk8s#1125]
**Component**: tests

**Why this is needed**: We require end-to-end upgrade tests.

**What should be done**:

Add a feature for upgrading a cluster in the BDD test suite, working locally.

* Input:
  * the URL for the artifacts of the origin build
* Behaviour:
  1. Initial installation with the origin build ISO
  2. Upgrade
* Output: a cluster in the expected, current version.

**Implementation proposal** (strongly recommended):

* Make this a `slow` test.
* Next steps to be kept in mind:
  * multi-node upgrade
  * Integration in CI

**Test plan**: This should respond to `tox -e tests -- -m "local and slow and upgrade"`

**Comment by TeddyAndrieux** [comment]
I think we should support iso as input instead of artifacts url.
I would say take path or link to an ISO as input.

**Comment by jbertran** [comment]
> I think we should support iso as input instead of artifacts url.
> I would say take path or link to an ISO as input.

Definitely, indeed using an artifacts URL would break things for anyone without access to the CI.

**Comment by Ebaneck** [comment]
Ideation Phase
```
@post @ci @local @slow
Feature: MetalK8s Upgrade
  As a Platform-Team Developer working on a 2.X feature
  I want to be able to connect to a ([2.x.x] -1) version of a Bootstrap node 
  And upgrade my Metalk8s cluster
  So I can confirm that the new feature I'm working on does not break the upgrade path.

  Scenario: Upgrade MetalK8s
    Given the Kubernetes API is available
    And I connect to the 'Bootstrap' node of a '([2.x.x] -1)' version
    And I get the current kubelet version on the Bootstrap node
    When I run the upgrade script with an Iso 'path'
    And I wait for the upgrade script to terminate
    Then the exit status of the upgrade script is '0'
    And the current kubelet version on the bootstrap node is greater than the previous
```

Ideas:

- Add a parallel eve step to spin up a cluster with a metalk8s version less than the feature branch version. OR An upgrade step to spin-up a cluster with version '([2.x.x] -1)' .

why?
We generally upgrade from one minor version to the other. So to test the upgrade, we need a cluster which is one minor version beneath the version of the feature branch.

Resource consumption?
We certainly will take up more extra openstack VM's => Problematic

Needs?

We need a pre-computed list of promoted metalk8s iso images such that our eve step knows
exactly which Metalk8s cluster image to spin-up based on the version number.

Ideas of which image to spin-up a metalk8s cluster have been raised such as using the last merged build but no conclusion on this... Needs more brainstorming
